source/model/Device.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+
+
+class Device(ABC):
+    cntr = 0
+
+    def __init__(self):
+        self.deviceId = self.__class__.newId()
+        self._activators = list()
+
+    @classmethod
+    def newId(cls):
+        cls.cntr += 1
+        return cls.cntr
+
+    @property
+    @abstractmethod
+    def name(self):
+        """ The name. So that we might identify different type of devices """
```

Comment:
You mean different products for the same type? Because you already have a pluginType method to identify different type of devices.